src/robotFactory.js

```diff
@@ -1,10 +1,86 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(value = 1) {
+    this.coords.y += value;
+  };
+
+  goBack(value = 1) {
+    this.coords.y -= value;
+  };
+
+  goLeft(value = 1) {
+    this.coords.x -= value;
+  };
+
+  goRight(value = 1) {
+    this.coords.x += value;
+  };
+
+  getInfo() {
+    return `Robot: ${this.name},`
+    + ` Chip version: ${this.chipVersion},`
+    + ` Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+      z: coords.z || 0,
+    };
+  }
+
+  goUp(value = 1) {
+    this.coords.z += value;
+  }
+
+  goDown(value = 1) {
+    this.coords.z -= value;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+    maxLoadWeight,
+    currentLoad
+  ) {
+    super(name, weight, coords, chipVersion);
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if (!this.currentLoad
+      && cargo.weight <= this.maxLoadWeight) {
+      this.currentLoad = cargo;
+    }
```

Comment:
```suggestion
    if (!this.currentLoad && cargo.weight <= this.maxLoadWeight) {
      this.currentLoad = cargo;
    }
```
guess the length is less than 80 :)